components/Products.tsx

```diff
@@ -42,7 +47,24 @@ const FETCH_STORE = gql`
   }
 `
 
-const NFT = ({ media, title }: { media: string; title: string }) => {
+const useBuy = (tokenID: string, tokenPrice: string) => {
+  const { wallet } = useWallet();
+  const tokenPriceNumber = Number(tokenPrice) ;
+  tokenPrice = (tokenPriceNumber).toLocaleString('fullwide', {useGrouping:false})
+  const buy = () => {
+    // create marketAddress env variable for testnet/mainnet
+    // wallet?.makeOffer(tokenID,tokenPrice,{ marketAddress: process.env.marketAddress})
+    wallet?.makeOffer(tokenID,tokenPrice)
+  }
+  return buy;
+}
+
+const NFT = ({ media, title, tokens }: { media: string; title: string; tokens: Token[] }) => {
+  //const NFT = ({ media, title }: { media: string; title: string}) => {
+  const { wallet, isConnected, details } = useWallet();
+
```

Comment:
I would do something like this:


```ts
const token = tokens[0]
const list = token?.list.price

// Maybe check if list is not null

const buy = useBuy(tokens[0]['id'],tokens[0].list.price) ;

```

Reply:
So this is one of the things I am not very sure about, meaning I have been told on Stackoverflow that you should not use hooks conditionally. Hence I don't wanna put useBuy() inside an if. However this won't break cause if you look at Query on line: 22. I have put the condition "where: { storeId: { _eq: $storeId }, burnedAt: { _is_null: true }, list: {removedAt: {_is_null: true}}}"
So NFTs which are not listed won't show up only. I am interested in doing this in a better way as well.

Reply:
I'm sorry, I may have not explained myself correctly. You're correct, ideally you would use a hook conditionally. 

I meant that the conditional logic should be inside the hook. Thinking again about this and I propose changing the hook to take just a `thingId` argument for now and resolve all the information inside of it. 

I think this query might help with that:

```ts
  query GetMarketListedTokens($thingId: String) {
    thing(where: { id: { _eq: $thingId } }, limit: 1) {
      metadata {
        title
        description
      }
      tokens(
        where: {
          _and: [
            { list: { autotransfer: { _eq: true } } }
            { list: { removedAt: { _is_null: true } } }
            { list: { createdAt: { _is_null: false } } }
          ]
        }
        order_by: { list: { price: asc } }
      ) {
        id
        list {
          id
          price
        }
      }
    }
  }
```

With this query you're essentially fetching all tokens that are listed in the market of one thing.

Rule of thumb is: Store -> Thing -> Tokens

Reply:
const buy = useBuy(thingId: string)
This makes a lot of sense, condition inside the hook is better than rendering it conditionally. I think if tokens array has length=0, that means there is no token of that thing listed. Do I need to use the above query?  But I think I can very easily incorporate the parameters you are using the above query in my own. Thanks. This make a lot of sense.